client/landing/gutenboarding/onboarding-block/signup-form/index.tsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import React, { useState } from 'react';
+
+// /**
+//  * Internal dependencies
+//  */
+import {
+	usePasswordlessSignUp,
+	UsePasswordlessSignUpStatus,
+	Provider,
+	Client,
+} from '@automattic/authentication';
+import { Button, TextControl, Modal } from '@wordpress/components';
+import FormLabel from 'components/forms/form-label';
+
+import './style.scss';
+
+const SignupForm = () => {
+	const [ emailVal, setEmailVal ] = useState( '' );
+	const { signUp, status, error } = usePasswordlessSignUp();
+
+	const handleSignUp = ( event: React.FormEvent< HTMLFormElement > ) => {
+		event.preventDefault();
+		signUp( emailVal );
+	};
+
+	const renderTosLink = () => {
+		return (
+			<a href="https://wordpress.com/tos/" target="_blank" rel="noopener noreferrer">
+				Terms of Service
+			</a>
+		);
+	};
+
+	return (
+		<Modal className="signup-form" title="Sign up to save your changes" onRequestClose={ () => {} }>
+			<form onSubmit={ handleSignUp }>
+				<FormLabel htmlFor="email">Your Email Address</FormLabel>
+				<TextControl
+					id="email"
+					value={ emailVal }
+					onChange={ setEmailVal }
+					placeholder="yourname@email.com"
+				/>
```

Comment:
Maybe add a `type="email"` prop here so that the correct keyboard appears on tablets?